notebooks/working/factor_models.py

```diff
@@ -0,0 +1,239 @@
+# +
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from causalgraphicalmodels import CausalGraphicalModel, StructuralCausalModel
+import pylogit
+from collections import OrderedDict
+import pylogit as cm
+from functools import reduce
+import statsmodels.api as sm
+import statsmodels.formula.api as smf
+from math import ceil
+
+from IPython import display
+import seaborn as sns
+
+import numpy as np
+import numpy.random as npr
+import pandas as pd
+import tensorflow as tf
+import tensorflow_probability as tfp
+import statsmodels.api as sm
+
+from tensorflow_probability import edward2 as ed
+from sklearn.datasets import load_breast_cancer
+from pandas.plotting import scatter_matrix
+from scipy import sparse, stats
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LogisticRegression 
+from sklearn.metrics import confusion_matrix, classification_report, roc_auc_score, roc_curve
+
+import matplotlib
+matplotlib.rcParams.update({'font.sans-serif' : 'Helvetica',
+                            'axes.labelsize': 10,
+                            'xtick.labelsize' : 6,
+                            'ytick.labelsize' : 6,
+                            'axes.titlesize' : 10})
+import matplotlib.pyplot as plt
+
+import seaborn as sns
+color_names = ["windows blue",
+               "amber",
+               "crimson",
+               "faded green",
+               "dusty purple",
+               "greyish"]
+colors = sns.xkcd_palette(color_names)
+sns.set(style="white", palette=sns.xkcd_palette(color_names), color_codes = False)
+
+
+from factor_models import *
```

Comment:
What is this line doing? Are you importing all of your code into yourself? This seems to be a very circular reference that I don't understand at all. Why is `factor_models.py` import `factor_models`?

Reply:
@timothyb0912 see latest commit where I clean the non-used imports. Removed this as well as some others.